cmd/checkup-launcher.go

```diff
@@ -47,6 +47,11 @@ func main() {
 	if err := logCheckupJobLogs(clientset, checkupJob.Namespace); err != nil {
 		log.Printf("Failed to dump checkup job logs: %v", err)
 	}
+	defer func() {
```

Comment:
Please consider removing `defer` usage here, its not clear what will happen since there are few `log.Fatalf` from line 56,
Also change the error handling of `Teardown` to not exit on error, logging the error is good enouge.